### PR TITLE
fix bug on kimi2.5 when dp2 tp4

### LINKED ISSUE #18410: [Bug] Kimi 2.5 has a chance of crashing under TP4 DP2 configuration.
### Checklist

- [x] I searched related issues but found no solution.
- [x] The bug persists in the latest version.
- [x] Issues without environment info and a minimal reproducible demo are hard to resolve and may receive no feedback.
- [x] If this is not a bug report but a general question, please start a discussion at https://github.com/sgl-project/sglang/discussions. Otherwise, it will be closed.
- [x] Please use English. Otherwise, it will be closed.

### Describe the bug

When making requests with data from test_error_ds.jsonl, the following error occurs:

Image

Image

test_error_ds.jsonl：
通过网盘分享的文件：test_error_ds.jsonl
链接: https://pan.baidu.com/s/1XPTQB9yq7l0D-4tO-Zyn0g 提取码: dtmy

### Reproduction

# sglang version:
commit 1552aab741bf89c241b10e06297ee657597bd758 (origin/community-main, community/main, community-main)
Author: 赵晨阳 <zhaochen20@outlook.com>
Date:   Fri Feb 6 20:33:29 2026 -0800

    Support execute_shell_command for env var support (#18390)

# bug reproduce:
1p1d, in 8*b200 machine.
start command for prefill node:
```
export CUDA_LAUNCH_BLOCKING=1;
export MODEL_PATH=/Kimi-K2.5;
export MASTER_IP=$(cut -d ',' -f 1 $FED_MEMBER_FILE);
export NUM_NODES=1;
export MC_TE_METRIC=true;
export SGLANG_TBO_DEBUG=1;
echo "Model Path: $MODEL_PATH"
echo "Master IP: $MASTER_IP"
python -m sglang.launch_server \
    --model-path $MODEL_PATH \
    --disaggregation-ib-device mlx5_bond_0 \
    --disaggregation-mode prefill \
    --disaggregation-bootstrap-port 8200 \
    --dist-init-addr $MASTER_IP:6676 \
    --nnodes $NUM_NODES \
    --node-rank $FED_POD_INDEX \
    --tp-size 8 \
    --dp-size 2 \
    --enable-dp-attention \
    --decode-log-interval 1 \
    --page-size 64 \
    --host 0.0.0.0 --port 8000 \
    --trust-remote-code \
    --moe-dense-tp-size 1 \
    --enable-dp-lm-head \
    --deepep-mode normal \
    --disable-cuda-graph \
    --mem-fraction-static 0.835 \
    --max-running-requests 256 \
    --max-total-tokens 700000 \
    --tool-call-parser kimi_k2 \
    --served-model-name $MODEL_ID \
    --enable-metrics --enable-cache-report --collect-tokens-histogram \
    --tokenizer-worker-num 8 \
    --log-level debug \
    --sampling-backend pytorch \
    --disable-overlap-schedule 
```
start command for decode node:
```
export MODEL_PATH=/Kimi-K2.5
export MASTER_IP=$(cut -d ',' -f 1 $FED_MEMBER_FILE)
export MC_TE_METRIC=true
export SGLANG_TBO_DEBUG=0
export NUM_NODES=1
export DP_SIZE=$((NUM_NODES * 8))
python -m sglang.launch_server \
--model-path ${MODEL_PATH} \
--disaggregation-ib-device mlx5_bond_0 \
--disaggregation-mode decode \
--prefill-round-robin-balance \
--dist-init-addr ${MASTER_IP}:6676 \
--nnodes $NUM_NODES \
--node-rank $FED_POD_INDEX \
--tp-size $DP_SIZE \
--dp-size $DP_SIZE \
--enable-dp-attention \
--decode-log-interval 2 \
--page-size 64 \
--host 0.0.0.0 --port 8000 \
--trust-remote-code \
--moe-dense-tp-size 1 \
--disable-radix-cache \
--watchdog-timeout 1000000 \
--log-level info \
--deepep-mode low_latency \
--mem-fraction-static 0.835 \
--max-total-tokens 700000 \
--max-running-requests $((256 * DP_SIZE)) \
--cuda-graph-max-bs 256 \
--tool-call-parser kimi_k2 \
--num-reserved-decode-tokens 102 \
--enable-metrics --enable-cache-report --collect-tokens-histogram \
--enable-dp-lm-head
```

### Environment

CUDA available: True
GPU 0,1,2,3,4,5,6,7: B200
GPU 0,1,2,3,4,5,6,7 Compute Capability: 10.0
CUDA_HOME: /usr/local/cuda
NVCC: Cuda compilation tools, release 12.9, V12.9.86
CUDA Driver Version: 580.95.05
PyTorch: 2.9.1+cu129
sglang: 0.5.8.post1
sgl_kernel: 0.3.21
flashinfer_python: 0.6.1
flashinfer_cubin: 0.6.1
flashinfer_jit_cache: 0.6.1+cu129
triton: 3.5.1
transformers: 4.57.1
torchao: 0.9.0
numpy: 2.4.2
aiohttp: 3.13.3
fastapi: 0.128.1
hf_transfer: 0.1.9
huggingface_hub: 0.36.1
interegular: 0.3.3
modelscope: 1.34.0
orjson: 3.11.7
outlines: 0.1.11
packaging: 26.0
psutil: 7.2.2
pydantic: 2.12.5
python-multipart: 0.0.22
pyzmq: 27.1.0
uvicorn: 0.40.0
uvloop: 0.22.1
vllm: Module Not Found
xgrammar: 0.1.27
openai: 2.6.1
tiktoken: 0.12.0
anthropic: 0.77.1
litellm: Module Not Found
decord2: 3.0.0
NVIDIA Topology: 
        GPU0    GPU1    GPU2    GPU3    GPU4    GPU5    GPU6    GPU7    NIC0    NIC1    NIC2    NIC3    NIC4    NIC5    NIC6    NIC7    NIC8    CPU Affinity  NUMA Affinity    GPU NUMA ID
GPU0     X      NV18    NV18    NV18    NV18    NV18    NV18    NV18    PIX     NODE    NODE    NODE    SYS     SYS     SYS     SYS     NODE    0-47,96-143   0N/A
GPU1    NV18     X      NV18    NV18    NV18    NV18    NV18    NV18    NODE    PIX     NODE    NODE    SYS     SYS     SYS     SYS     NODE    0-47,96-143   0N/A
GPU2    NV18    NV18     X      NV18    NV18    NV18    NV18    NV18    NODE    NODE    PIX     NODE    SYS     SYS     SYS     SYS     NODE    0-47,96-143   0N/A
GPU3    NV18    NV18    NV18     X      NV18    NV18    NV18    NV18    NODE    NODE    NODE    PIX     SYS     SYS     SYS     SYS     NODE    0-47,96-143   0N/A
GPU4    NV18    NV18    NV18    NV18     X      NV18    NV18    NV18    SYS     SYS     SYS     SYS     PIX     NODE    NODE    NODE    SYS     48-95,144-191 1N/A
GPU5    NV18    NV18    NV18    NV18    NV18     X      NV18    NV18    SYS     SYS     SYS     SYS     NODE    PIX     NODE    NODE    SYS     48-95,144-191 1N/A
GPU6    NV18    NV18    NV18    NV18    NV18    NV18     X      NV18    SYS     SYS     SYS     SYS     NODE    NODE    PIX     NODE    SYS     48-95,144-191 1N/A
GPU7    NV18    NV18    NV18    NV18    NV18    NV18    NV18     X      SYS     SYS     SYS     SYS     NODE    NODE    NODE    PIX     SYS     48-95,144-191 1N/A
NIC0    PIX     NODE    NODE    NODE    SYS     SYS     SYS     SYS      X      NODE    NODE    NODE    SYS     SYS     SYS     SYS     NODE
NIC1    NODE    PIX     NODE    NODE    SYS     SYS     SYS     SYS     NODE     X      NODE    NODE    SYS     SYS     SYS     SYS     NODE
NIC2    NODE    NODE    PIX     NODE    SYS     SYS     SYS     SYS     NODE    NODE     X      NODE    SYS     SYS     SYS     SYS     NODE
NIC3    NODE    NODE    NODE    PIX     SYS     SYS     SYS     SYS     NODE    NODE    NODE     X      SYS     SYS     SYS     SYS     NODE
NIC4    SYS     SYS     SYS     SYS     PIX     NODE    NODE    NODE    SYS     SYS     SYS     SYS      X      NODE    NODE    NODE    SYS
NIC5    SYS     SYS     SYS     SYS     NODE    PIX     NODE    NODE    SYS     SYS     SYS     SYS     NODE     X      NODE    NODE    SYS
NIC6    SYS     SYS     SYS     SYS     NODE    NODE    PIX     NODE    SYS     SYS     SYS     SYS     NODE    NODE     X      NODE    SYS
NIC7    SYS     SYS     SYS     SYS     NODE    NODE    NODE    PIX     SYS     SYS     SYS     SYS     NODE    NODE    NODE     X      SYS
NIC8    NODE    NODE    NODE    NODE    SYS     SYS     SYS     SYS     NODE    NODE    NODE    NODE    SYS     SYS     SYS     SYS      X 

Legend:

  X    = Self
  SYS  = Connection traversing PCIe as well as the SMP interconnect between NUMA nodes (e.g., QPI/UPI)
  NODE = Connection traversing PCIe as well as the interconnect between PCIe Host Bridges within a NUMA node
  PHB  = Connection traversing PCIe as well as a PCIe Host Bridge (typically the CPU)
  PXB  = Connection traversing multiple PCIe bridges (without traversing the PCIe Host Bridge)
  PIX  = Connection traversing at most a single PCIe bridge
  NV#  = Connection traversing a bonded set of # NVLinks

NIC Legend:

  NIC0: mlx5_4
  NIC1: mlx5_7
  NIC2: mlx5_8
  NIC3: mlx5_9
  NIC4: mlx5_10
  NIC5: mlx5_11
  NIC6: mlx5_12
  NIC7: mlx5_13
  NIC8: mlx5_bond_0

ulimit soft: 1048576

### PR DESCRIPTION
**Pull request by haowen-han**
## Motivation
This bug fix was a joint effort by me and my colleague @engineer1109 . Thanks to @engineer1109 for the collaboration!

fix https://github.com/sgl-project/sglang/issues/18410

## Modifications

torch.empty to torch.zeros, so nan will not appear for pad tokens

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

**Review by mickqian — APPROVED** [review]

**Comment by mickqian** [comment]
irrelevant flaky ci testcase failed, bypassing
